integration/utils/try.go

```diff
@@ -1,50 +1,209 @@
+func TryResponse(req *http.Request, timeout time.Duration) (*http.Response, error) {
+	return doResponse(req, timeout, nil)
+}
+
+// TryResponseUntilStatusCode is like TryRequest, but returns the response for further
+// processing at the call site.
+// Conditions are not allowed since it would complicate signaling if the
+// response body needs to be closed or not. Callers are expected to close on
+// their own if the function returns a nil error.
+func TryResponseUntilStatusCode(req *http.Request, timeout time.Duration, statusCode int) (*http.Response, error) {
+	return doResponse(req, timeout, UntilStatusCodeIs(statusCode))
+}
+
+// TryGetRequest is like Try, but runs a request against the given URL and applies
+// the condition on the response.
+// Condition may be nil, in which case only the request against the URL must
+// succeed.
+func TryGetRequest(url string, timeout time.Duration, condition Condition) error {
+	resp, err := doGetResponse(url, timeout, condition)
+
+	if resp != nil && resp.Body != nil {
+		defer resp.Body.Close()
+	}
+
 	return err
 }
 
-// Try try operation timeout, and retry backoff
-func Try(timeout time.Duration, operation func() error) error {
-	exponentialBackOff := backoff.NewExponentialBackOff()
-	exponentialBackOff.MaxElapsedTime = timeout
-	err := backoff.Retry(operation, exponentialBackOff)
+// TryRequest is like Try, but runs a request against the given URL and applies
+// the condition on the response.
+// Condition may be nil, in which case only the request against the URL must
+// succeed.
+func TryRequest(req *http.Request, timeout time.Duration, condition Condition) error {
+	resp, err := doResponse(req, timeout, condition)
+
+	if resp != nil && resp.Body != nil {
+		defer resp.Body.Close()
+	}
+
 	return err
 }
 
+// Try repeatedly executes an operation until no error condition occurs or the
+// given timeout is reached, whatever comes first.
+func Try(timeout time.Duration, operation func() error) error {
+	if timeout <= 0 {
+		panic("timeout must be larger than zero")
+	}
+
+	interval := time.Duration(math.Ceil(float64(timeout) / 15.0))
+	if interval > maxInterval {
+		interval = maxInterval
+	}
+
+	ci := os.Getenv("CI")
+	if len(ci) > 0 {
+		log.Println("Activate CI multiplier:", CITimeoutMultiplier)
+		timeout = time.Duration(float64(timeout) * CITimeoutMultiplier)
+	}
+
+	var err error
+	if err = operation(); err == nil {
+		return nil
+	}
+
+	stopTime := time.Now().Add(timeout)
+
+	for {
+		if time.Now().After(stopTime) {
+			fmt.Println("-")
+			return fmt.Errorf("try operation failed: %s", err)
+		}
+
+		// FIXME refactor this part
+		select {
+		case <-time.Tick(interval):
+			fmt.Print("*")
+			if err = operation(); err == nil {
+				fmt.Println("+")
+				return nil
+			}
+		}
+	}
+}
+
+// Sleep pauses the current goroutine for at least the duration d.
+// Use only when use an other Try[...] functions is not possible.
+func Sleep(d time.Duration) {
```

Comment:
https://github.com/containous/traefik/blob/master/integration/marathon_test.go#L39

This case need a `continue` approach, `try` doesn't work in this case.

PS: this test file seems to test nothing, because all request seems always give the same response, look at the comment block (it's impossible to make this work).